examples/custom-metrics/src/main/java/com/spotify/zoltar/examples/metrics/CustomMetricsExample.java

```diff
@@ -0,0 +1,186 @@
+      };
+    }
+
+    @Override
+    public FeatureExtractorMetrics<Integer, Float> featureExtractorMetrics() {
+      return id -> {
+        final CustomMetrics metrics = metricsCache().getUnchecked(id);
+        final Counter negativeExtractCounter = metrics.negativeExtractCount();
+        return NegativeExtractMetrics.create(negativeExtractCounter);
+      };
+    }
+  }
+
+  /**
+   * To define a metric you want to measure for prediction, implement PredictMetrics.
+   */
+  @AutoValue
+  abstract static class NegativePredictMetrics implements PredictMetrics<Integer, Float> {
+    abstract Counter negativePredictCount();
+
+    static NegativePredictMetrics create(final Counter negativeCount) {
+      return new AutoValue_CustomMetricsExample_NegativePredictMetrics(negativeCount);
+    }
+
+    /**
+     * Here, given a list of predictions, we want to measure the number of negatives.
+     */
+    @Override
+    public void prediction(final List<Prediction<Integer, Float>> predictions) {
+      negativePredictCount().inc(predictions.stream().filter(x -> x.value() < 0).count());
+    }
+  }
+
+  /**
+   * To define a metric you want to measure for feature extraction, implement VectorMetrics.
+   */
+  @AutoValue
+  abstract static class NegativeExtractMetrics implements VectorMetrics<Integer, Float> {
+    abstract Counter negativeExtractCount();
+
+    static NegativeExtractMetrics create(final Counter negativeCount) {
+      return new AutoValue_CustomMetricsExample_NegativeExtractMetrics(negativeCount);
+    }
+
+    /**
+     * Here, given a list of feature extraction results, we want to count the number of negatives.
+     */
+    @Override
+    public void extraction(final List<Vector<Integer, Float>> vectors) {
+      negativeExtractCount().inc(vectors.stream().filter(v -> v.value() < 0).count());
+    }
+  }
+
+  CustomMetricsExample(final SemanticMetricRegistry metricRegistry, final MetricId metricId)
+      throws Exception {
+    final ModelLoader<DummyModel> modelLoader = ModelLoader.lift(DummyModel::new);
+    final SingleExtractFn<Integer, Float> extractFn = input -> (float) input / 10;
+    final PredictFn<DummyModel, Integer, Float, Float> predictFn = (model, vectors) -> {
+      return vectors.stream()
+          .map(vector -> Prediction.create(vector.input(),vector.value() * 2))
+          .collect(Collectors.toList());
+    };
+
+    // We build the PredictorBuilder as usual, compose with the built-in metrics, and then compose
+    // with our custom metrics.
+    predictorBuilder = Predictors.newBuilder(
+        modelLoader,
+        FeatureExtractor.create(extractFn),
+        predictFn
+    ).with(Instrumentations.predictor(SemanticPredictorMetrics.create(metricRegistry, metricId))
+    ).with(Instrumentations.predictor(CustomPredictorMetrics.create(metricRegistry, metricId)));
+
+    final DummyModel model = modelLoader.get(Duration.ofMinutes(1));
```

Comment:
@andrewsmartin instead of doing this you can:

```java
    final ModelLoader<DummyModel> modelLoader = ModelLoader
        .lift(DummyModel::new)
        .with(Preloader.preload(Duration.ofMinutes(1)));
```

Reply:
Nice!